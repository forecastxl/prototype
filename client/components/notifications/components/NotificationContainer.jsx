import React, { PropTypes } from 'react'
import { connect } from 'react-redux'
import { getNotifications } from '../selectors'
import NotificationBox from './NotificationBox'
import Notification from './Notification'

export function NotificationContainer({ notifications }) {
  if (notifications.length === 0) {
    return null
  }

  return (
    <NotificationBox>
      {notifications.map(notification => (
        <Notification key={notification}>
          {notification}
        </Notification>
      ))}
    </NotificationBox>
  )
}

NotificationContainer.propTypes = {
  notifications: PropTypes.array.isRequired
}

const mapStateToProps = state => ({
  notifications: getNotifications(state)
})

export default connect(mapStateToProps)(NotificationContainer)
