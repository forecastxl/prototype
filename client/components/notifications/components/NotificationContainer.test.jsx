import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { NotificationContainer } from './NotificationContainer'

describe('<NotificationContainer />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<NotificationContainer notifications={['message']} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('renders correctly when there are no notifications', () => {
    const wrapper = shallow(<NotificationContainer notifications={[]} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})

