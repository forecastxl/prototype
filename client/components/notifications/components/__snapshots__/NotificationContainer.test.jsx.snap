// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationContainer /> renders correctly 1`] = `
<styled.div>
  <styled.div>
    message
  </styled.div>
</styled.div>
`;

exports[`<NotificationContainer /> renders correctly when there are no notifications 1`] = `null`;
